core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -287,7 +287,8 @@ private[storage] class BlockManagerDecommissioner(
     val livePeerSet = bm.getPeers(false).toSet
     val currentPeerSet = migrationPeers.keys.toSet
     val deadPeers = currentPeerSet.diff(livePeerSet)
-    val newPeers = livePeerSet.diff(currentPeerSet)
+    // Randomize the orders of the peers to avoid migrating data to same nodes
+    val newPeers = Utils.randomize(livePeerSet.diff(currentPeerSet))
```

Comment:
`livePeerSet.diff(currentPeerSet)` only returns new nodes. It doesn't seem to we'd migrate data to the same node here, right?

Reply:
It'll most likely migrate to the same set of nodes as:
1. `bm.getPeers()` always return sorted collection of peers. So if two node start decommission at the close time, they will get the same ordered set of peers.
2. When `currentPeerSet` is empty at the beginning, the `newPeers` is still the same order.

Reply:
I see. You're talking about the orders for two different decommissioning nodes.